presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1726,7 +1731,14 @@ else if (probeSymbols.contains(new SymbolReference(secondVariable.getName())) &&
 
         private SpatialPredicate spatialTest(CallExpression functionCall, boolean probeFirst, Optional<OperatorType> comparisonOperator)
         {
-            QualifiedFunctionName functionName = metadata.getFunctionManager().getFunctionMetadata(functionCall.getFunctionHandle()).getName();
+            FunctionMetadata functionMetadata = metadata.getFunctionManager().getFunctionMetadata(functionCall.getFunctionHandle());
+            QualifiedFunctionName functionName = functionMetadata.getName();
+
+            if (functionMetadata.getArgumentTypes().stream().allMatch((typeSignature)
+                    -> typeSignature.equals(SPHERICAL_GEOGRAPHY_TYPE_SIGNATURE))) {
+                return sphericalSpatialTest(functionName, comparisonOperator);
+            }
+
```

Comment:
If everything that follows is euclidian instead of spherical then stuff those into a sub-function the same as you did for sphericalSpatialTest.

Also do we care to handle mixed signatures here (some spherical, some not?). What is the right behavior there?